### ISSUE TITLE
TODO: Re-implement Commented Out Light Code

### ISSUE THREAD

**Issue by Fronkln** [lewa-j/Unity-Source-Tools#35]
I recently have re-implemented commented out lighting code to recreate lighting entities and have it only apply to objects that are **not a part of the map lightmap** the results are very nice. Should be worth to have it added to this

Before:
![image](https://user-images.githubusercontent.com/17107072/65388254-90552000-dd54-11e9-8028-bf2604bc0cbf.png)
After:
![image](https://user-images.githubusercontent.com/17107072/65388305-a7940d80-dd54-11e9-8016-9f4aa3434dc2.png)

Note: the bloom effect comes from Post-Processing Stack v2

**Comment by lewa-j** [comment]
SaracenOne already did that
https://github.com/SaracenOne/Unity-Source-Tools

**Comment by Fronkln** [comment]
That applies to everything on scene though, it's not gonna look pretty, lightmaps are gonna become even brighter

**Comment by Fronkln** [comment]
Addittionally, i have written extra code regarding this that will help with spotlights as well

**Comment by Fronkln** [comment]
Another thing i have noticed, it "assumes" certain values of lights, not trying to make the best out of the data provided by BSP so it's more flexible

**Comment by Fronkln** [comment]
The fact that SaracenOne's applies to the entire map is **VERY** unoptimized as well, i'm guessing lights would constantly flicker (due to pixel light size and that the lights apply to **ENTIRE** map)

Let's assume that he raised the pixel light count to 60, at this point the game will start lagging due to calculating so much lights at once, his method can work on pre-loaded maps but it will absolutely **KILL** optimization if used at real-time

Now let's take a look at my method:

Light's are only applied to a few select layers (and never the entire map!) so unless there is like 30-60 entities wandering around the entire map, light flickers (Unity transitioning pixel light to vertex light) can **NEVER** be noticeable, my method also doesn't need to raise the pixel light count to disgusting amounts (like 30-60) and can work with 2-6.

Hope that clears it up.

**Comment by SaracenOne** [comment]
You're quite right about it being woefully unoptimised, but if I recall, I was enabling them with a very specific purpose, which was to simply use them for someone baking out entirely new lightmaps and probes from scratch.
